src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/Policy.kt

```diff
@@ -0,0 +1,129 @@
+        return toXContent(builder, ToXContent.EMPTY_PARAMS)
+    }
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        builder
+            .startObject()
+                .startObject(POLICY_TYPE)
+                    .field(NAME_FIELD, name)
+                    .optionalTimeField(LAST_UPDATED_TIME_FIELD, lastUpdatedTime)
+                    .field(SCHEMA_VERSION_FIELD, schemaVersion)
+                    .field(DEFAULT_NOTIFICATION_FIELD, defaultNotification)
+                    .field(DEFAULT_STATE_FIELD, defaultState)
+                    .field(STATES_FIELD, states.toTypedArray())
+                .endObject()
+            .endObject()
+        return builder
+    }
+
+    companion object {
+        const val POLICY_TYPE = "policy"
+        const val NAME_FIELD = "name"
+        const val NO_ID = ""
+        const val NO_VERSION = Versions.NOT_FOUND
+        const val LAST_UPDATED_TIME_FIELD = "last_updated_time"
+        const val SCHEMA_VERSION_FIELD = "schema_version"
+        const val DEFAULT_NOTIFICATION_FIELD = "default_notification"
+        const val DEFAULT_STATE_FIELD = "default_state"
+        const val STATES_FIELD = "states"
+
+        @JvmStatic
+        @JvmOverloads
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser, id: String = NO_ID, version: Long = NO_VERSION): Policy {
+            lateinit var name: String
+            lateinit var defaultState: String
+            // TODO Implement DefaultNotification(destination, message)
+            var defaultNotification: Map<String, Any>? = null
+            var lastUpdatedTime: Instant? = null
+            var schemaVersion: Long = 1
+            val states: MutableList<Map<String, Any>> = mutableListOf() // TODO: MutableList<State>
+
+            ensureExpectedToken(Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
+            while (xcp.nextToken() != Token.END_OBJECT) {
+                val fieldName = xcp.currentName()
+                xcp.nextToken()
+
+                when (fieldName) {
+                    SCHEMA_VERSION_FIELD -> schemaVersion = xcp.longValue()
+                    LAST_UPDATED_TIME_FIELD -> lastUpdatedTime = xcp.instant()
+                    NAME_FIELD -> name = xcp.text()
+                    // TODO: DefaultNotification.parse(xcp)
+                    DEFAULT_NOTIFICATION_FIELD -> defaultNotification = null
+                    DEFAULT_STATE_FIELD -> defaultState = xcp.text()
+                    STATES_FIELD -> {
+                        ensureExpectedToken(Token.START_ARRAY, xcp.currentToken(), xcp::getTokenLocation)
+                        while (xcp.nextToken() != Token.END_ARRAY) {
+                            // TODO: states.add(State.parse(xcp))
+                        }
+                    }
+                }
+            }
+
+            ensureExpectedToken(Token.END_OBJECT, xcp.nextToken(), xcp::getTokenLocation)
```

Comment:
Is this considered redundant since the ```while``` exit condition is ```Token.END_OBJECT```? If not, we may want to add this to [ManagedIndex.kt](https://github.com/opendistro-for-elasticsearch/index-management/blob/development/src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/ManagedIndex.kt) as well

Reply:
Good catch, meant for the parseWithType for that last END_OBJET when we're parsing { "policy": { ... } }